exercises/queen-attack/example.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Debug, PartialEq)]
+pub struct Queen {
+    position: (i8, i8)
+}
+
+#[derive(Debug, PartialEq)]
+pub struct InvalidQueen;
+
+impl Queen {
+    pub fn new(position: (i8, i8)) -> Result<Queen, InvalidQueen> {
+        if (position.0 < 0 || position.0 > 7) || (position.1 < 0 || position.1 >7 ) {
+            Err(InvalidQueen)
+        } else {
+            Ok(Queen {position: position})
+        }
+    }
+}
+
+pub struct Queens {
+    first: Queen,
+    second: Queen
+}
+
+impl Queens {
+    pub fn new(first: Result<Queen, InvalidQueen>, second: Result<Queen, InvalidQueen>) -> Queens {
```

Comment:
I'm thinking a bit about the interface here; seems a bit odd to have those `Result<Queen, InvalidQueen>` as the argument types.

What if it were `(i8, i8)` here instead of the `Result<Queen, InvalidQueen>`, and returning `Result<Queens, InvalidQueen>` (or maybe `InvalidQueens`? Or just `Option<Queens>`?)

I can understand why the `Result<Queen, InvalidQueen>` is done (so that the result of `Queen::new` can be directly passed in, and then `Queen::new` can be tested individually instead of `Queens::new` which has more decision-making to do). Just seems an unclear interface.

What if it were just `queen1.can_attack(queen2)` ?

I could be convinced either way on this, feel free to discuss it while I'm traveling.